src/d3a/models/market/balancing.py

```diff
@@ -51,13 +51,15 @@ def __init__(self, time_slot=None, bc=None, notification_listener=None, readonly
 
     def offer(self, price: float, energy: float, seller: str, offer_id=None,
               original_offer_price=None, dispatch_event=True, seller_origin=None,
-              adapt_price_with_fees=True, seller_origin_id=None):
+              adapt_price_with_fees=True, seller_origin_id=None,
+              attributes: dict = None, requirements: List[Dict] = None):
```

Comment:
Nit: as I pointed out in the other PR (on d3a-api-client), the bid/offer methods are using too many arguments. This means that we probably can/should figure out a way to refactor them and use a more OOP-friendly approach. The risk of keeping adding new arguments is that the complexity will increase, it'll be hard to extend and refactor the methods, and testing them will become a nightmare.

Reply:
For example, `seller`, `seller_origin` and `seller_origin_id` conceptually belong to the same entity (I would call it `Seller`). The rest of the parameters might belong to a `OfferParameter` class, but this is  just a quick idea.